### PR TITLE
Automatic update of swversion

### PR DESCRIPTION
**Pull request by panovvv**
* Now clicking the "Install update" button in Hue app will actually work (tested by running the emulator code locally).
* Parses https://www.philips-hue.com/en-us/support/release-notes/bridge and pulls the latest released version from there.
* This new feature is controlled by a flag (commandline argument/env. variable) that's switched off by default.

### CODE REVIEW

File: BridgeEmulator/HueEmulator3.py [hunk 1]
```diff
@@ -1577,20 +1585,33 @@ def read_http_request_body(self):
         return b"{}" if self.headers['Content-Length'] is None or self.headers[
             'Content-Length'] == '0' else self.rfile.read(int(self.headers['Content-Length']))
 
+    @staticmethod
+    def extract_version(line: str):
+        if match := re.search(r'Firmware( |(&nbsp;))+([0-9]+).*Bridge( |(&nbsp;))+V2', line, re.IGNORECASE):
```
Comment (panovvv) [inline BridgeEmulator/HueEmulator3.py:1590]: Flexible regex - case insensitive, whitespace between words can be any combination of spaces and &nbsp; codes.

File: BridgeEmulator/HueEmulator3.py [hunk 2]
```diff
@@ -1577,20 +1585,33 @@ def read_http_request_body(self):
...
+            return int(match.group(3))
+
     def do_POST(self):
         self._set_headers()
         logging.info("in post method")
         logging.info(self.path)
-        self.data_string = self.read_http_request_body()
```
Comment (panovvv) [inline BridgeEmulator/HueEmulator3.py:1584]: We're ignoring request body in /updater code, but reading it takes considerable time (the application sends the whole binary file in it). Therefore I'm only reading the request body in all cases except the /updater.

**Comment by Mevel** [comment]
Really nice! I suggest triggering a config backup when pushing the update button or new version is found while parsing hue website.

**Comment by panovvv** [comment]
Sure thing! I'll try to figure out how it's done now....

**Comment by panovvv** [comment]
@Mevel  okay, I'll try to proceed with the approach I've described above...

**Comment by panovvv** [comment]
@Mevel  all done!

**Comment by alexyao2015** [comment]
https://github.com/diyhue/diyHue/blob/17552e0ca928b642c665fe43653e6fa48c00bff0/src/core/configManager/configOperations.py#L216
FYI

**Comment by panovvv** [comment]
@alexyao2015 Cool! This is from the `experimental` branch, right? When are you planning on merging it? If not (or the planned merge is far in the future) I can carry that code over to this branch instead. This feature would be really useful - kinda sucks having each Philips update brick the whole system until you adjust the configuration...

**Comment by alexyao2015** [comment]
Yes thats the experimental branch. Its already implemented there so there isn't anything to port there.

**Pull request closed without merging**